qcodes/instrument_drivers/Keysight/MSOS104A.py

```diff
@@ -0,0 +1,368 @@
+        channel_data.fromstring(data[2:-1]) #skip first 2 bytes and last byte
+        #channel_data.byteswap()
+        #channel_data.reverse()
+        channel_data = np.array(channel_data)
+        channel_data = np.multiply(channel_data, yinc) + yorigin
+
+        oscil.read_termination = term
+
+        return channel_data
+
+    def _setup_instrument_for_acquisition(self):
+        instr = self._instrument
+        instr.write(":waveform:format WORD")
+        instr.write(":waveform:byteorder LSBFirst")
+        instr._parent.acquire_mode('RTIMe')
+        # instr.write(":waveform:streaming 1")
+
+class MSOChannel(InstrumentChannel):
+
+    def __init__(self, parent, name, channel):
+        super().__init__(parent, name)
+        # scaling
+        self.add_parameter(name='offset',
+                           label='Channel {} offset'.format(channel),
+                           set_cmd='CHAN{}:OFFS {{}}'.format(channel),
+                           unit='V',
+                           get_cmd='CHAN{}:OFFS?'.format(channel),
+                           get_parser=float
+                           )
+
+        self.add_parameter(name='scale',
+                           label='Channel {} scale'.format(channel),
+                           unit='V/div',
+                           set_cmd='CHAN{}:SCAL {{}}'.format(channel),
+                           get_cmd='CHAN{}:SCAL?'.format(channel),
+                           get_parser=float,
+                           vals=vals.Numbers(0,100)  # TODO: upper limit?
+                           )
+
+        self.add_parameter(name='range',
+                           label='Channel {} range'.format(channel),
+                           unit='V',
+                           set_cmd='CHAN{}:RANG {{}}'.format(channel),
+                           get_cmd='CHAN{}:RANG?'.format(channel),
+                           get_parser=float,
+                           vals=vals.Numbers()
+                           )
+        # trigger
+        self.add_parameter('trigger_level',
+                           label='Tirgger level channel {}'.format(channel),
+                           unit='V',
+                           get_cmd=':TRIGger:LEVel? CHANnel{}'.format(channel),
+                           set_cmd=\
+                               ':TRIGger:LEVel CHANnel{},{{}}'.format(channel),
+                           get_parser=float,
+                           vals=Numbers(),
+                          )
+
+        # Acquisition
+        self.add_parameter( name='trace',
+                            channel = channel,
+                            parameter_class=RawTrace
+                            )
+        def snapshot_base(self, update: bool=False) -> Dict:
+            params_to_skip_update = ['trace']
+            super().snapshot_base(update=update,
+                                  params_to_skip_update=params_to_skip_update)
+
+
+class MSOS104A(VisaInstrument):
```

Comment:
I think it should have a more generic name the only difference that i can tell is the max frequency